cxx/pclean/pclean_lib_test.cc

```diff
@@ -0,0 +1,57 @@
+#define BOOST_TEST_MODULE test pclean_csv
+
+#include "pclean/pclean_lib.hh"
+#include <sstream>
+#include <boost/test/included/unit_test.hpp>
+namespace tt = boost::test_tools;
+
+BOOST_AUTO_TEST_CASE(test_translate_observations) {
+  std::stringstream ss(R"""(Column1,Room Type,Monthly Rent,County,State
+0,studio,,Mahoning County,OH
+1,4br,2152.0,,NV
+2,1br,1267.0,Gwinnett County,
+)""");
+
+  DataFrame df = DataFrame::from_csv(ss);
+
+  std::map<std::string, std::string> state_params = {{"strings", "AL AK AZ AR CA CO CT DE DC FL GA HI ID IL IN IA KS KY LA ME MD MA MI MN MS MO MT NE NV NH NJ NM NY NC ND OH OK OR PA RI SC SD TN TX UT VT VA WA WV WI WY"}};
+  std::map<std::string, std::string> br_params = {{"strings", "1br 2br 3br 4br studio"}};
+
+  T_schema schema = {
+    {"County:name",
+      T_clean_relation{{"County"}, false, DistributionSpec("bigram")}},
+    {"County:state",
+      T_clean_relation{{"County"}, false, DistributionSpec("stringcat", state_params)}},
+    {"Room Type",
+      T_clean_relation{{"Obs"}, false, DistributionSpec("stringcat", br_params)}},
+    {"Monthly Rent",
+      T_clean_relation{{"Obs"}, false, DistributionSpec("normal")}},
+    {"County",
```

Comment:
For clarity, could you give the "County" and "State" relations names that disambiguate them from the domains? These names would be output from the schema converter as "Obs:county:name" and "Obs:state:name", right?

Reply:
I've prepended a "d" to all of the domain names for clarity.

The schema converter would give these relation names of "County" and "State", actually -- for noisy relations coming from an "observe x as y" line, the relation name is always taken as the "y", so that it can match the csv column name.